### ISSUE TITLE
E12 team to provide custom permissions table

### ISSUE THREAD

**Issue by anchit-chandran** [rcpch/rcpch-audit-engine#544]
Update docs perms table once provided

**Comment by eatyourpeas** [comment]
These map roles to groups.
| Role | Group |
|------|------|
| Audit Centre Lead Clinician | trust_audit_team_view_only |
| Audit Centre Clinician | trust_audit_team_edit_access |
| Audit Centre Administrator | trust_audit_team_full_access |
| RCPCH Audit Team | epilepsy12_audit_team_full_access |
| RCPCH Audit Children and Family | patient_access |

Basic permissions are already mapped to groups and agreed
These custom permissions have not yet been agreed:
```python
CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING
CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING
CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_AUDIT
CAN_APPROVE_ELIGIBILITY
CAN_REGISTER_CHILD_IN_EPILEPSY12
CAN_UNREGISTER_CHILD_IN_EPILEPSY12
CAN_EDIT_EPILEPSY12_LEAD_CENTRE
CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE
CAN_TRANSFER_EPILEPSY12_LEAD_CENTRE
CAN_DELETE_EPILEPSY12_LEAD_CENTRE
CAN_CONSENT_TO_AUDIT_PARTICIPATION
```

@AmaniKrayemRCPCH @nikyraja could you please allocate permissions to groups?

**Comment by AmaniKrayemRCPCH** [comment]
Heya,

Firstly, check if the table you've given is correct? Should these groups be swapped
<img width="301" alt="Capture1" src="https://github.com/rcpch/rcpch-audit-engine/assets/126471383/25eb66a9-6fc4-4cfc-a374-3c4698d17495">

Will add a comment with permissions shortly

**Comment by AmaniKrayemRCPCH** [comment]
to confirm, by 'Register' you mean to add a new patient? If so, the below is correct. 

Group | Permissions
--- | ---
trust_audit_team_view_only | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br> CAN_REGISTER_CHILD_IN_EPILEPSY12
trust_audit_team_edit_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br>  CAN_APPROVE_ELIGIBILITY<br> CAN_REGISTER_CHILD_IN_EPILEPSY12<br>  CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE
trust_audit_team_full_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br>  CAN_APPROVE_ELIGIBILITY<br> CAN_REGISTER_CHILD_IN_EPILEPSY12<br>  CAN_UNREGISTER_CHILD_IN_EPILEPSY12<br> CAN_EDIT_EPILEPSY12_LEAD_CENTRE<br>  CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE<br>  CAN_TRASNFER_EPILEPSY12_LEAD_CENTRE<br>  CAN_DELETE_EPILEPSY12_LEAD_CENTRE
epilepsy12_audit_team_full_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br>  CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br>  CAN_APPROVE_ELIGIBILITY<br> CAN_REGISTER_CHILD_IN_EPILEPSY12<br>  CAN_UNREGISTER_CHILD_IN_EPILEPSY12<br> CAN_EDIT_EPILEPSY12_LEAD_CENTRE<br>  CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE<br>  CAN_TRASNFER_EPILEPSY12_LEAD_CENTRE<br>  CAN_DELETE_EPILEPSY12_LEAD_CENTRE
patient_access | CAN_CONSENT_TO_AUDIT_PARTICIPATION

**Comment by AmaniKrayemRCPCH** [comment]
sorry, slight edit to the trust_audit_team_view_only permissions

Group | Permissions
-- | --
trust_audit_team_view_only | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_REGISTER_CHILD_IN_EPILEPSY12
trust_audit_team_edit_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_APPROVE_ELIGIBILITYCAN_REGISTER_CHILD_IN_EPILEPSY12CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE
trust_audit_team_full_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_APPROVE_ELIGIBILITYCAN_REGISTER_CHILD_IN_EPILEPSY12CAN_UNREGISTER_CHILD_IN_EPILEPSY12CAN_EDIT_EPILEPSY12_LEAD_CENTRECAN_ALLOCATE_EPILEPSY12_LEAD_CENTRECAN_TRASNFER_EPILEPSY12_LEAD_CENTRECAN_DELETE_EPILEPSY12_LEAD_CENTRE
epilepsy12_audit_team_full_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_APPROVE_ELIGIBILITYCAN_REGISTER_CHILD_IN_EPILEPSY12CAN_UNREGISTER_CHILD_IN_EPILEPSY12CAN_EDIT_EPILEPSY12_LEAD_CENTRECAN_ALLOCATE_EPILEPSY12_LEAD_CENTRECAN_TRASNFER_EPILEPSY12_LEAD_CENTRECAN_DELETE_EPILEPSY12_LEAD_CENTRE
patient_access | CAN_CONSENT_TO_AUDIT_PARTICIPATION

**Comment by eatyourpeas** [comment]
Thank you
By register, I mean complete the registration. That is not the same as creating a case – it involves agreeing eligibility, setting the FPA date and confirming the lead centre. It maybe this permission is not needed but currently it exists and is allocated as listed

S

From: Amani Krayem ***@***.***>
Date: Thursday, 24 August 2023 at 09:26
To: rcpch/rcpch-audit-engine ***@***.***>
Cc: Simon Chapman ***@***.***>, Comment ***@***.***>
Subject: Re: [rcpch/rcpch-audit-engine] E12 team to provide custom permissions table (Issue #544)

sorry, slight edit to the trust_audit_team_view_only permissions
Group
Permissions
trust_audit_team_view_only
CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_REGISTER_CHILD_IN_EPILEPSY12
trust_audit_team_edit_access
CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_APPROVE_ELIGIBILITYCAN_REGISTER_CHILD_IN_EPILEPSY12CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE
trust_audit_team_full_access
CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_APPROVE_ELIGIBILITYCAN_REGISTER_CHILD_IN_EPILEPSY12CAN_UNREGISTER_CHILD_IN_EPILEPSY12CAN_EDIT_EPILEPSY12_LEAD_CENTRECAN_ALLOCATE_EPILEPSY12_LEAD_CENTRECAN_TRASNFER_EPILEPSY12_LEAD_CENTRECAN_DELETE_EPILEPSY12_LEAD_CENTRE
epilepsy12_audit_team_full_access
CAN_LOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITINGCAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDITCAN_APPROVE_ELIGIBILITYCAN_REGISTER_CHILD_IN_EPILEPSY12CAN_UNREGISTER_CHILD_IN_EPILEPSY12CAN_EDIT_EPILEPSY12_LEAD_CENTRECAN_ALLOCATE_EPILEPSY12_LEAD_CENTRECAN_TRASNFER_EPILEPSY12_LEAD_CENTRECAN_DELETE_EPILEPSY12_LEAD_CENTRE
patient_access
CAN_CONSENT_TO_AUDIT_PARTICIPATION

**Comment by AmaniKrayemRCPCH** [comment]
Ah understood. Ok, the correct version of the permissions table is below

Group | Permissions
-- | --
trust_audit_team_view_only | 
trust_audit_team_edit_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br> CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br> CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br> CAN_APPROVE_ELIGIBILITY<br> CAN_REGISTER_CHILD_IN_EPILEPSY12<br> CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE
trust_audit_team_full_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br> CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br> CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br> CAN_APPROVE_ELIGIBILITY<br> CAN_REGISTER_CHILD_IN_EPILEPSY12<br> CAN_UNREGISTER_CHILD_IN_EPILEPSY12<br> CAN_EDIT_EPILEPSY12_LEAD_CENTRE<br> CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE<br> CAN_TRASNFER_EPILEPSY12_LEAD_CENTRE<br> CAN_DELETE_EPILEPSY12_LEAD_CENTRE
epilepsy12_audit_team_full_access | CAN_LOCK_CHILD_CASE_DATA_FROM_EDITING<br> CAN_UNLOCK_CHILD_CASE_DATA_FROM_EDITING<br> CAN_OPT_OUT_CHILD_FROM_INCLUSION_IN_THE_AUDIT<br> CAN_APPROVE_ELIGIBILITY<br> CAN_REGISTER_CHILD_IN_EPILEPSY12<br> CAN_UNREGISTER_CHILD_IN_EPILEPSY12<br> CAN_EDIT_EPILEPSY12_LEAD_CENTRE<br> CAN_ALLOCATE_EPILEPSY12_LEAD_CENTRE<br> CAN_TRASNFER_EPILEPSY12_LEAD_CENTRE<br> CAN_DELETE_EPILEPSY12_LEAD_CENTRE
patient_access | CAN_CONSENT_TO_AUDIT_PARTICIPATION

**Comment by AmaniKrayemRCPCH** [comment]
Can I check on the status of this? Has it been actioned?

Is this related to #541

**Comment by eatyourpeas** [comment]
This still needs implementing

**Comment by eatyourpeas** [comment]
closed with #640
